### PR TITLE
Crafting GUI utility adjustments (batch savings, uniform skill requirement display)

### PR DESCRIPTION
**Pull request by Asmageddon**
The changed interface looks like this:
image

As before, it displays respectively `N/A` and `none` for lack of primary/other skill requirements, and `none` for lack of batch savings.

Includes a typo fix. Tested by scrolling through most recipes in the game, no breakage observed.

**EDIT**: Resolves https://github.com/CleverRaven/Cataclysm-DDA/issues/27669

SUMMARY: Interface "Crafting UI enhancements (batch time, skill reqs)"

### CODE REVIEW

File: src/crafting_gui.cpp [hunk 1]
```diff
@@ -555,26 +555,22 @@ const recipe *select_crafting_recipe( int &batch_size )
...
-                    string_format( _( "Your skill level: <color_%s>%d</color>" ),
-                                   difficulty_color, player_skill ) :
-                    _( "Your skill level: <color_yellow>N/A</color>" );
-                print_colored_text( w_data, ypos++, xpos, col, col, skill_level_string );
+                    string_format( _( "Primary skill: <color_cyan>%s</color> <color_%s>%s</color>" ),
+                                   ( !current[line]->skill_used ? _( "N/A" ) :
+                                     current[line]->skill_used.obj().name() ),
+                                   difficulty_color,
+                                   ( !current[line]->skill_used ? "" : primary_skill_level )
+                                 ));
```
Comment (Night-Pryanik) [inline src/crafting_gui.cpp]:
```suggestion
                                 ) );
```

File: src/crafting_gui.cpp [hunk 2]
```diff
@@ -583,6 +579,10 @@ const recipe *select_crafting_recipe( int &batch_size )
                                         _( "Time to complete: <color_cyan>%s</color>" ),
                                         to_string( time_duration::from_turns( expected_turns ) ) );
 
+                ypos += fold_and_print( w_data, ypos, xpos, pane, col,
+                                        _( "Batch time savings: <color_cyan>%s</color>" ),
+                                        current[line]->batch_savings_string());
```
Comment (Night-Pryanik) [inline src/crafting_gui.cpp]:
```suggestion
                                        current[line]->batch_savings_string() );
```

File: src/recipe.cpp [hunk 3]
```diff
@@ -386,11 +386,25 @@ std::string recipe::required_skills_string( const Character *c ) const
     [&]( const std::pair<skill_id, int> &skill ) {
         auto player_skill = c ? c->get_skill_level( skill.first ) : 0;
         std::string difficulty_color = skill.second > player_skill ? "yellow" : "green";
-        return string_format( "<color_cyan>%s</color> <color_%s>(%d)</color>",
-                              skill.first.obj().name(), difficulty_color, skill.second );
+        std::string skill_level_string = print_skill_level ? "" : (std::to_string(player_skill) + "/");
+        skill_level_string += std::to_string(skill.second);
```
Comment (Night-Pryanik) [inline src/recipe.cpp]:
```suggestion
        skill_level_string += std::to_string( skill.second );
```

File: src/recipe.cpp [hunk 4]
```diff
@@ -386,11 +386,25 @@ std::string recipe::required_skills_string( const Character *c ) const
     [&]( const std::pair<skill_id, int> &skill ) {
         auto player_skill = c ? c->get_skill_level( skill.first ) : 0;
         std::string difficulty_color = skill.second > player_skill ? "yellow" : "green";
-        return string_format( "<color_cyan>%s</color> <color_%s>(%d)</color>",
-                              skill.first.obj().name(), difficulty_color, skill.second );
+        std::string skill_level_string = print_skill_level ? "" : (std::to_string(player_skill) + "/");
```
Comment (Night-Pryanik) [inline src/recipe.cpp]:
```suggestion
        std::string skill_level_string = print_skill_level ? "" : ( std::to_string( player_skill ) + "/" );
```

File: src/recipe.cpp [hunk 5]
```diff
@@ -386,11 +386,25 @@ std::string recipe::required_skills_string( const Character *c ) const
...
 }
 
+std::string recipe::required_skills_string( const Character *c ) const
+{
+    return required_skills_string(c, false);
+}
+
+std::string recipe::batch_savings_string () const
+{
+    return (batch_rsize != 0) ?
```
Comment (Night-Pryanik) [inline src/recipe.cpp]:
```suggestion
    return ( batch_rsize != 0 ) ?
```

File: src/recipe.cpp [hunk 6]
```diff
@@ -386,11 +386,25 @@ std::string recipe::required_skills_string( const Character *c ) const
...
+        string_format( _( "%s%% at >%s units" ), int(batch_rscale * 100), batch_rsize)
```
Comment (Night-Pryanik) [inline src/recipe.cpp]:
```suggestion
        string_format( _( "%s%% at >%s units" ), int( batch_rscale * 100 ), batch_rsize )
```

**Comment by ZhilkinSerg** [comment]
Jenkins rebuild.

**Comment by AskaHope** [comment]
Can't you add a "food heating" quality or something? Every tool use the same amount of charges, isn't there a way to simplify this?

**Comment by kevingranade** [comment]
Squashed and merged to 0.D and master.

**Pull request closed without merging**